easybuild/easyblocks/o/openblas.py

```diff
@@ -3,18 +3,42 @@
 
 @author: Andrew Edmondson (University of Birmingham)
 """
+import os
 from easybuild.easyblocks.generic.configuremake import ConfigureMake
-from easybuild.tools.systemtools import POWER, get_cpu_architecture
+from easybuild.tools.systemtools import POWER, get_cpu_architecture, get_shared_lib_ext
 
 
 class EB_OpenBLAS(ConfigureMake):
     """Support for building/installing OpenBLAS."""
 
-    def build_step(self):
-        """Custom build procedure for OpenBLAS."""
+    def configure_step(self):
+        """ set up some options - but no configure command to run"""
+
+        if 'BINARY=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'BINARY=64')
+        if 'USE_THREAD=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'USE_THREAD=1')
+            self.cfg.update('installopts', 'USE_THREAD=1')
+        if 'USE_OPENMP=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'USE_OPENMP=1')
+            self.cfg.update('installopts', 'USE_OPENMP=1')
+        if 'CC=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'CC=%s' % os.environ['CC'])
+        if 'FC=' not in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'FC=%s' % os.environ['FC'])
 
         if get_cpu_architecture() == POWER:
             # There doesn't seem to be a POWER9 option yet, but POWER8 should work.
-            self.cfg['buildopts'] += ' TARGET=POWER8'
+            self.cfg.update('buildopts', 'TARGET=POWER8')
```

Comment:
Not sure if this is relevant for `installopts`, but maybe it's better to join this with the above, as follows (before the `for` loop):

```python
if get_cpu_architecture() == POWER:
    default_opts['TARGET'] = 'POWER8'

Reply:
Done